packages/dds/map/src/mapFactory.ts

```diff
@@ -9,19 +9,19 @@ import type {
 	IFluidDataStoreRuntime,
 	IChannelServices,
 } from "@fluidframework/datastore-definitions/internal";
-import { createSharedObjectKind } from "@fluidframework/shared-object-base/internal";
+import {
+	createSharedObjectKind,
+	type ISharedObject,
+} from "@fluidframework/shared-object-base/internal";
 
 import type { ISharedMap } from "./interfaces.js";
 import { SharedMap as SharedMapInternal } from "./map.js";
 import { pkgVersion } from "./packageVersion.js";
 
 /**
  * {@link @fluidframework/datastore-definitions#IChannelFactory} for {@link ISharedMap}.
- *
- * @sealed
- * @alpha
  */
-export class MapFactory implements IChannelFactory<ISharedMap> {
+export class MapFactory implements IChannelFactory<ISharedMap & ISharedObject> {
```

Comment:
This has been removed from the package API as users of it should use `SharedMap.getFactory()` instead. Doing this removal allows it to adopt types more friendly to its own internal unit testing.

Reply:
This is not related to your PR, and maybe too late to have such discussion. But I think the preferred direction is for objects not to know about their factories, and only factory to know how to create objects.
This could be useful in composition, where there could be many different factory types that create same object, but with different policies (for example, schemas or type of schemas, in case of shared tree).

Reply:
This code is in full agreement with your statement. This factory, which already had a direct dependency on the type is produces continues to do so. The object does not use the factory anymore: I removed those statics in previous work when using ISharedObjectKind instead of class exports for the API entry point. This change does not impact that aspect either.